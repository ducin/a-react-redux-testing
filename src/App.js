import React from 'react';
import './App.css';
import { EmployeeList } from './components/employees/EmployeeList';

import { getEmployees } from './data'
const employees = getEmployees().slice(0, 25)

function App({ all = true }) {
  // const sum = calculateData(all)
  const sum = 125
  return (
    <div className="App">
      elo: {sum}
      <EmployeeList employees={employees} />
    </div>
  );
}

export default App;
