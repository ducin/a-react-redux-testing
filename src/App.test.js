import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import App from "./App";

it("should render without crashing #smoke", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("should render without crashing (with prop all=false) #smoke", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App all={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("should match snapshot", () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
