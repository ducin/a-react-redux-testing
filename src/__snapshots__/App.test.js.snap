// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="App"
>
  elo: 
  125
  <ol>
    <li>
      Bertram
       
      Kruszewski
      , 
      Junior .Net Engineer
    </li>
    <li>
      John
       
      Herzog
      , 
      Principal .Net Developer
    </li>
    <li>
      Leila
       
      Szeląg
      , 
      Principal Software Developer
    </li>
    <li>
      Jacopo
       
      Costa
      , 
      .Net Developer
    </li>
    <li>
      Matteo
       
      Martino
      , 
      Junior .Net Developer
    </li>
    <li>
      Conchita
       
      Granados
      , 
      Senior .Net Engineer
    </li>
    <li>
      Dorothea
       
      Biesenbach
      , 
      Junior Big Data Developer
    </li>
    <li>
      Amy
       
      Lenzen
      , 
      Software Engineer
    </li>
    <li>
      Agrypina
       
      Czapla
      , 
      Software Engineer
    </li>
    <li>
      Greta
       
      Zając
      , 
      Software Engineer
    </li>
    <li>
      Rodrigo
       
      Ebert
      , 
      Senior Software Developer
    </li>
    <li>
      Erik
       
      Senger
      , 
      Junior Big Data Engineer
    </li>
    <li>
      Ferne
       
      Denesik
      , 
      Junior .Net Engineer
    </li>
    <li>
      Elody
       
      Kohler
      , 
      Software Developer
    </li>
    <li>
      Roman
       
      Buczkowski
      , 
      Java Developer
    </li>
    <li>
      Władysław
       
      Olejniczak
      , 
      Junior .Net Developer
    </li>
    <li>
      Ansgary
       
      Ruciński
      , 
      Junior .Net Engineer
    </li>
    <li>
      Sander
       
      Bruin
      , 
      Junior Software Engineer
    </li>
    <li>
      Lisa
       
      Veen
      , 
      Junior Software Developer
    </li>
    <li>
      Mikołaj
       
      Sochacki
      , 
      Principal Java Engineer
    </li>
    <li>
      Carmelo
       
      Kris
      , 
      Senior Java Engineer
    </li>
    <li>
      Stanisław
       
      Mikulski
      , 
      Big Data Developer
    </li>
    <li>
      Ruben
       
      Smits
      , 
      .Net Engineer
    </li>
    <li>
      Josefina
       
      Olivares
      , 
      Senior .Net Developer
    </li>
    <li>
      Georg
       
      Mesloh
      , 
      Junior Java Developer
    </li>
    <li>
      Luis Miguel
       
      Salcedo
      , 
      Junior Java Developer
    </li>
    <li>
      Fritz
       
      Corwin
      , 
      Java Engineer
    </li>
    <li>
      Amira
       
      Einert
      , 
      Senior Java Engineer
    </li>
    <li>
      Clemens
       
      Trantow
      , 
      Big Data Developer
    </li>
    <li>
      Zoe
       
      Prevost
      , 
      Big Data Developer
    </li>
    <li>
      Dario
       
      Hentschke
      , 
      Senior Java Engineer
    </li>
    <li>
      Eriberto
       
      Upton
      , 
      Big Data Engineer
    </li>
    <li>
      Maurycy
       
      Żurawski
      , 
      Senior Java Engineer
    </li>
    <li>
      Giorgio
       
      Barbieri
      , 
      Junior Big Data Engineer
    </li>
    <li>
      Alice
       
      Beer
      , 
      Software Developer
    </li>
    <li>
      Doriana
       
      Barbieri
      , 
      Senior .Net Developer
    </li>
    <li>
      Mark
       
      Neuschwander
      , 
      Software Developer
    </li>
    <li>
      Krystel
       
      Ferry
      , 
      Java Engineer
    </li>
    <li>
      Bertrand
       
      Cholewa
      , 
      Big Data Developer
    </li>
    <li>
      Fernando
       
      Salgado
      , 
      Senior Software Engineer
    </li>
    <li>
      Carmen
       
      Collazo
      , 
      Big Data Developer
    </li>
    <li>
      Kevin
       
      Jacobs
      , 
      Senior Big Data Engineer
    </li>
    <li>
      Max
       
      Piras
      , 
      Junior Software Engineer
    </li>
    <li>
      Kira
       
      Jamrozy
      , 
      Junior Java Engineer
    </li>
    <li>
      Zachary
       
      Frączek
      , 
      .Net Engineer
    </li>
    <li>
      Klaudiusz
       
      Woś
      , 
      Junior Software Developer
    </li>
    <li>
      Shawn
       
      Sanford
      , 
      Senior Big Data Engineer
    </li>
    <li>
      Otha
       
      Wiegand
      , 
      Big Data Engineer
    </li>
    <li>
      Lucjan
       
      Radecki
      , 
      Senior Big Data Engineer
    </li>
    <li>
      Cecylia
       
      Janiszewski
      , 
      .Net Engineer
    </li>
    <li>
      Jole
       
      Riva
      , 
      Senior Big Data Developer
    </li>
    <li>
      Piotr
       
      Gołąb
      , 
      Senior Java Developer
    </li>
    <li>
      Lilou
       
      Fournier
      , 
      Software Developer
    </li>
    <li>
      Teodora
       
      Florek
      , 
      Principal Software Developer
    </li>
    <li>
      Eliasz
       
      Kwiatkowski
      , 
      Principal .Net Engineer
    </li>
    <li>
      Maryam
       
      Voigt
      , 
      Senior Big Data Developer
    </li>
    <li>
      Arystarch
       
      Woźniak
      , 
      Junior Big Data Engineer
    </li>
    <li>
      Herta
       
      Hermiston
      , 
      .Net Developer
    </li>
    <li>
      Marianna
       
      Kaleta
      , 
      Software Developer
    </li>
    <li>
      Olaf
       
      Kmiecik
      , 
      Big Data Developer
    </li>
    <li>
      Lucja
       
      Godlewski
      , 
      .Net Engineer
    </li>
    <li>
      Xenia
       
      Baumann
      , 
      Junior Big Data Developer
    </li>
    <li>
      Davion
       
      Bruen
      , 
      Senior .Net Engineer
    </li>
    <li>
      Bertram
       
      Jóźwiak
      , 
      Senior .Net Engineer
    </li>
    <li>
      Jonatan
       
      Konieczna
      , 
      Principal Software Engineer
    </li>
    <li>
      Delilah
       
      Gleason
      , 
      Big Data Developer
    </li>
    <li>
      Tarik
       
      Schmidt
      , 
      Software Engineer
    </li>
    <li>
      Elżbieta
       
      Dąbek
      , 
      Software Developer
    </li>
    <li>
      Dell
       
      Fahey
      , 
      Senior Software Developer
    </li>
    <li>
      Laurencja
       
      Chrzanowski
      , 
      .Net Developer
    </li>
    <li>
      Roman
       
      Frątczak
      , 
      Principal .Net Developer
    </li>
    <li>
      Meagan
       
      Steuber
      , 
      Senior Java Developer
    </li>
    <li>
      Kevin
       
      Jong
      , 
      Senior .Net Developer
    </li>
    <li>
      Candelario
       
      Homenick
      , 
      Principal .Net Engineer
    </li>
    <li>
      Amber
       
      Haan
      , 
      Software Engineer
    </li>
    <li>
      Niko
       
      Verniest
      , 
      Senior Software Engineer
    </li>
    <li>
      Kim
       
      Sauer
      , 
      .Net Developer
    </li>
    <li>
      Władysław
       
      Witkowski
      , 
      .Net Developer
    </li>
    <li>
      Bette
       
      Towne
      , 
      Junior Java Developer
    </li>
    <li>
      Anastazja
       
      Gałązka
      , 
      Java Developer
    </li>
    <li>
      Rudy
       
      Bernardi
      , 
      .Net Developer
    </li>
    <li>
      Ewa
       
      Kaźmierczak
      , 
      Junior .Net Developer
    </li>
    <li>
      Faustino
       
      O'Conner
      , 
      Junior Software Developer
    </li>
    <li>
      Lisa
       
      Smit
      , 
      Principal .Net Developer
    </li>
    <li>
      Sanne
       
      Bosch
      , 
      Senior .Net Developer
    </li>
    <li>
      Victor
       
      Lucas
      , 
      Junior Big Data Engineer
    </li>
    <li>
      Diamond
       
      Wolff
      , 
      Big Data Engineer
    </li>
    <li>
      Filemon
       
      Tkaczyk
      , 
      Senior Big Data Developer
    </li>
    <li>
      Marissa
       
      Jess
      , 
      Java Developer
    </li>
    <li>
      Mayra
       
      West
      , 
      Junior .Net Developer
    </li>
    <li>
      Leona
       
      Pflieger
      , 
      Senior Software Developer
    </li>
    <li>
      Mekhi
       
      Torphy
      , 
      .Net Engineer
    </li>
    <li>
      Marie
       
      Aubert
      , 
      Senior .Net Developer
    </li>
    <li>
      Lindsey
       
      Parisian
      , 
      Software Engineer
    </li>
    <li>
      Joachim
       
      Baranowski
      , 
      Junior .Net Developer
    </li>
    <li>
      Unique
       
      Champlin
      , 
      Principal .Net Developer
    </li>
    <li>
      Celeste
       
      Silvestri
      , 
      Junior Software Engineer
    </li>
    <li>
      Amira
       
      Casper
      , 
      Senior Big Data Developer
    </li>
    <li>
      Casimir
       
      Satterfield
      , 
      Junior Big Data Engineer
    </li>
    <li>
      Gastone
       
      Benedetti
      , 
      Senior Java Engineer
    </li>
  </ol>
  <hr />
  <div>
    <div>
      <span
        className="page"
        disabled={true}
        onClick={[Function]}
      >
         
        &lt;&lt;
         
      </span>
      <span
        className="page"
        disabled={true}
        onClick={[Function]}
      >
         &lt; 
      </span>
      <span
        className="page selected"
        onClick={[Function]}
      >
        1
      </span>
      <span
        className="page"
        onClick={[Function]}
      >
        2
      </span>
      <span
        className="page"
        onClick={[Function]}
      >
        3
      </span>
      <span
        className="page"
        disabled={false}
        onClick={[Function]}
      >
         &gt; 
      </span>
      <span
        className="page"
        disabled={false}
        onClick={[Function]}
      >
         
        &gt;&gt;
         
      </span>
    </div>
    <span>
      items per page
    </span>
    :
    <span
      className="selected"
      onClick={[Function]}
    >
      10
    </span>
     | 
    <span
      className=""
      onClick={[Function]}
    >
      20
    </span>
     | 
    <span
      className=""
      onClick={[Function]}
    >
      30
    </span>
    <ol>
      <li>
        Bertram
         
        Kruszewski
        , 
        Junior .Net Engineer
      </li>
      <li>
        John
         
        Herzog
        , 
        Principal .Net Developer
      </li>
      <li>
        Leila
         
        Szeląg
        , 
        Principal Software Developer
      </li>
      <li>
        Jacopo
         
        Costa
        , 
        .Net Developer
      </li>
      <li>
        Matteo
         
        Martino
        , 
        Junior .Net Developer
      </li>
      <li>
        Conchita
         
        Granados
        , 
        Senior .Net Engineer
      </li>
      <li>
        Dorothea
         
        Biesenbach
        , 
        Junior Big Data Developer
      </li>
      <li>
        Amy
         
        Lenzen
        , 
        Software Engineer
      </li>
      <li>
        Agrypina
         
        Czapla
        , 
        Software Engineer
      </li>
      <li>
        Greta
         
        Zając
        , 
        Software Engineer
      </li>
      <li>
        Rodrigo
         
        Ebert
        , 
        Senior Software Developer
      </li>
      <li>
        Erik
         
        Senger
        , 
        Junior Big Data Engineer
      </li>
      <li>
        Ferne
         
        Denesik
        , 
        Junior .Net Engineer
      </li>
      <li>
        Elody
         
        Kohler
        , 
        Software Developer
      </li>
      <li>
        Roman
         
        Buczkowski
        , 
        Java Developer
      </li>
      <li>
        Władysław
         
        Olejniczak
        , 
        Junior .Net Developer
      </li>
      <li>
        Ansgary
         
        Ruciński
        , 
        Junior .Net Engineer
      </li>
      <li>
        Sander
         
        Bruin
        , 
        Junior Software Engineer
      </li>
      <li>
        Lisa
         
        Veen
        , 
        Junior Software Developer
      </li>
      <li>
        Mikołaj
         
        Sochacki
        , 
        Principal Java Engineer
      </li>
      <li>
        Carmelo
         
        Kris
        , 
        Senior Java Engineer
      </li>
      <li>
        Stanisław
         
        Mikulski
        , 
        Big Data Developer
      </li>
      <li>
        Ruben
         
        Smits
        , 
        .Net Engineer
      </li>
      <li>
        Josefina
         
        Olivares
        , 
        Senior .Net Developer
      </li>
      <li>
        Georg
         
        Mesloh
        , 
        Junior Java Developer
      </li>
      <li>
        Luis Miguel
         
        Salcedo
        , 
        Junior Java Developer
      </li>
      <li>
        Fritz
         
        Corwin
        , 
        Java Engineer
      </li>
      <li>
        Amira
         
        Einert
        , 
        Senior Java Engineer
      </li>
      <li>
        Clemens
         
        Trantow
        , 
        Big Data Developer
      </li>
      <li>
        Zoe
         
        Prevost
        , 
        Big Data Developer
      </li>
      <li>
        Dario
         
        Hentschke
        , 
        Senior Java Engineer
      </li>
      <li>
        Eriberto
         
        Upton
        , 
        Big Data Engineer
      </li>
      <li>
        Maurycy
         
        Żurawski
        , 
        Senior Java Engineer
      </li>
      <li>
        Giorgio
         
        Barbieri
        , 
        Junior Big Data Engineer
      </li>
      <li>
        Alice
         
        Beer
        , 
        Software Developer
      </li>
      <li>
        Doriana
         
        Barbieri
        , 
        Senior .Net Developer
      </li>
      <li>
        Mark
         
        Neuschwander
        , 
        Software Developer
      </li>
      <li>
        Krystel
         
        Ferry
        , 
        Java Engineer
      </li>
      <li>
        Bertrand
         
        Cholewa
        , 
        Big Data Developer
      </li>
      <li>
        Fernando
         
        Salgado
        , 
        Senior Software Engineer
      </li>
      <li>
        Carmen
         
        Collazo
        , 
        Big Data Developer
      </li>
      <li>
        Kevin
         
        Jacobs
        , 
        Senior Big Data Engineer
      </li>
      <li>
        Max
         
        Piras
        , 
        Junior Software Engineer
      </li>
      <li>
        Kira
         
        Jamrozy
        , 
        Junior Java Engineer
      </li>
      <li>
        Zachary
         
        Frączek
        , 
        .Net Engineer
      </li>
      <li>
        Klaudiusz
         
        Woś
        , 
        Junior Software Developer
      </li>
      <li>
        Shawn
         
        Sanford
        , 
        Senior Big Data Engineer
      </li>
      <li>
        Otha
         
        Wiegand
        , 
        Big Data Engineer
      </li>
      <li>
        Lucjan
         
        Radecki
        , 
        Senior Big Data Engineer
      </li>
      <li>
        Cecylia
         
        Janiszewski
        , 
        .Net Engineer
      </li>
      <li>
        Jole
         
        Riva
        , 
        Senior Big Data Developer
      </li>
      <li>
        Piotr
         
        Gołąb
        , 
        Senior Java Developer
      </li>
      <li>
        Lilou
         
        Fournier
        , 
        Software Developer
      </li>
      <li>
        Teodora
         
        Florek
        , 
        Principal Software Developer
      </li>
      <li>
        Eliasz
         
        Kwiatkowski
        , 
        Principal .Net Engineer
      </li>
      <li>
        Maryam
         
        Voigt
        , 
        Senior Big Data Developer
      </li>
      <li>
        Arystarch
         
        Woźniak
        , 
        Junior Big Data Engineer
      </li>
      <li>
        Herta
         
        Hermiston
        , 
        .Net Developer
      </li>
      <li>
        Marianna
         
        Kaleta
        , 
        Software Developer
      </li>
      <li>
        Olaf
         
        Kmiecik
        , 
        Big Data Developer
      </li>
      <li>
        Lucja
         
        Godlewski
        , 
        .Net Engineer
      </li>
      <li>
        Xenia
         
        Baumann
        , 
        Junior Big Data Developer
      </li>
      <li>
        Davion
         
        Bruen
        , 
        Senior .Net Engineer
      </li>
      <li>
        Bertram
         
        Jóźwiak
        , 
        Senior .Net Engineer
      </li>
      <li>
        Jonatan
         
        Konieczna
        , 
        Principal Software Engineer
      </li>
      <li>
        Delilah
         
        Gleason
        , 
        Big Data Developer
      </li>
      <li>
        Tarik
         
        Schmidt
        , 
        Software Engineer
      </li>
      <li>
        Elżbieta
         
        Dąbek
        , 
        Software Developer
      </li>
      <li>
        Dell
         
        Fahey
        , 
        Senior Software Developer
      </li>
      <li>
        Laurencja
         
        Chrzanowski
        , 
        .Net Developer
      </li>
      <li>
        Roman
         
        Frątczak
        , 
        Principal .Net Developer
      </li>
      <li>
        Meagan
         
        Steuber
        , 
        Senior Java Developer
      </li>
      <li>
        Kevin
         
        Jong
        , 
        Senior .Net Developer
      </li>
      <li>
        Candelario
         
        Homenick
        , 
        Principal .Net Engineer
      </li>
      <li>
        Amber
         
        Haan
        , 
        Software Engineer
      </li>
      <li>
        Niko
         
        Verniest
        , 
        Senior Software Engineer
      </li>
      <li>
        Kim
         
        Sauer
        , 
        .Net Developer
      </li>
      <li>
        Władysław
         
        Witkowski
        , 
        .Net Developer
      </li>
      <li>
        Bette
         
        Towne
        , 
        Junior Java Developer
      </li>
      <li>
        Anastazja
         
        Gałązka
        , 
        Java Developer
      </li>
      <li>
        Rudy
         
        Bernardi
        , 
        .Net Developer
      </li>
      <li>
        Ewa
         
        Kaźmierczak
        , 
        Junior .Net Developer
      </li>
      <li>
        Faustino
         
        O'Conner
        , 
        Junior Software Developer
      </li>
      <li>
        Lisa
         
        Smit
        , 
        Principal .Net Developer
      </li>
      <li>
        Sanne
         
        Bosch
        , 
        Senior .Net Developer
      </li>
      <li>
        Victor
         
        Lucas
        , 
        Junior Big Data Engineer
      </li>
      <li>
        Diamond
         
        Wolff
        , 
        Big Data Engineer
      </li>
      <li>
        Filemon
         
        Tkaczyk
        , 
        Senior Big Data Developer
      </li>
      <li>
        Marissa
         
        Jess
        , 
        Java Developer
      </li>
      <li>
        Mayra
         
        West
        , 
        Junior .Net Developer
      </li>
      <li>
        Leona
         
        Pflieger
        , 
        Senior Software Developer
      </li>
      <li>
        Mekhi
         
        Torphy
        , 
        .Net Engineer
      </li>
      <li>
        Marie
         
        Aubert
        , 
        Senior .Net Developer
      </li>
      <li>
        Lindsey
         
        Parisian
        , 
        Software Engineer
      </li>
      <li>
        Joachim
         
        Baranowski
        , 
        Junior .Net Developer
      </li>
      <li>
        Unique
         
        Champlin
        , 
        Principal .Net Developer
      </li>
      <li>
        Celeste
         
        Silvestri
        , 
        Junior Software Engineer
      </li>
      <li>
        Amira
         
        Casper
        , 
        Senior Big Data Developer
      </li>
      <li>
        Casimir
         
        Satterfield
        , 
        Junior Big Data Engineer
      </li>
      <li>
        Gastone
         
        Benedetti
        , 
        Senior Java Engineer
      </li>
    </ol>
  </div>
</div>
`;
